### PR TITLE
curvefs/client: fix problem of perf decline

### PR DESCRIPTION
**Pull request by wuhongsong**
### What problem does this PR solve?

Issue Number: #1627  

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/client/s3/disk_cache_manager.cpp [hunk 1]
```diff
@@ -380,18 +380,19 @@ void DiskCacheManager::TrimCache() {
...
             // then read will fail when do cache read,
             // and then it cannot load the file from S3.
             // so read is fail.
-            if (ret == 0) {
+            while (ret == 0) {
                 VLOG(1) << "do not remove this disk file"
                         << ", file has not been uploaded to S3."
                         << ", file is: " << cacheKey;
-                continue;
+                bthread_usleep(1000);
```
Comment (xu-chaojie) [inline curvefs/src/client/s3/disk_cache_manager.cpp:394]: why use bthread_usleep？ this function is running in bthread？

**Comment by wuhongsong** [comment]
when file is deleted in cachewrite ,then we will call `cachedObjName_->Put(fileName, false)` ，and it will put the obj to the head of the list, so the  funciton GetLast will cost many  times， and it will  block flush because it will call `diskCacheManager_->AddCache(name)` .

File: curvefs/src/client/s3/disk_cache_manager.cpp [hunk 2]
```diff
@@ -159,13 +159,13 @@ int DiskCacheManager::ClearReadCache(const std::list<std::string> &files) {
 
 void DiskCacheManager::AddCache(const std::string name,
   bool cacheWriteExist) {
-    cachedObjName_->Put(name, cacheWriteExist);
+    cachedObjName_->Put(name);
     VLOG(9) << "cache size is: " << cachedObjName_->Size();
 }
 
 bool DiskCacheManager::IsCached(const std::string name) {
     bool exist;
```
Comment (xu-chaojie) [inline curvefs/src/client/s3/disk_cache_manager.cpp:167]: This variable is no longer used.

File: src/common/lru_cache.h [hunk 3]
```diff
@@ -551,6 +559,26 @@ void SglLRUCache<K, KeyTraits>::Put(const K &key) {
...
+    ll_.push_back(key);
+    cache_[key] = --ll_.end();
+    size_++;
+    if (cacheMetrics_ != nullptr) {
+        cacheMetrics_->UpdateAddToCacheCount();
+        cacheMetrics_->UpdateAddToCacheBytes(KeyTraits::CountBytes(key));
+    }
+    return true;
+}
+
```
Comment (baijiaruo) [inline src/common/lru_cache.h:581]: I think the objname of the current diskcache no longer belongs to lru, it is recommended to implement this module by yourself

Comment (wuhongsong) [reply src/common/lru_cache.h:583]: ok， i have add todo in the code, will edit later.

**Review by baijiaruo — APPROVED** [review]

**Review by xu-chaojie — APPROVED** [review]

**Pull request merged**